translator/trace/jaeger/jaegerproto_to_protospan.go

```diff
@@ -101,18 +101,21 @@ func jProtoSpansToOCProtoSpans(jspans []*model.Span) []*tracepb.Span {
 			TraceId: tracetranslator.UInt64ToByteTraceID(jspan.TraceID.High, jspan.TraceID.Low),
 			SpanId:  tracetranslator.UInt64ToByteSpanID(uint64(jspan.SpanID)),
 			// TODO: Tracestate: Check RFC status and if is applicable,
-			ParentSpanId: tracetranslator.UInt64ToByteSpanID(uint64(jspan.ParentSpanID())),
-			Name:         strToTruncatableString(jspan.OperationName),
-			Kind:         sKind,
-			StartTime:    internal.TimeToTimestamp(jspan.StartTime),
-			EndTime:      internal.TimeToTimestamp(jspan.StartTime.Add(jspan.Duration)),
-			Attributes:   sAttributes,
+			Name:       strToTruncatableString(jspan.OperationName),
+			Kind:       sKind,
+			StartTime:  internal.TimeToTimestamp(jspan.StartTime),
+			EndTime:    internal.TimeToTimestamp(jspan.StartTime.Add(jspan.Duration)),
+			Attributes: sAttributes,
 			// TODO: StackTrace: OpenTracing defines a semantic key for "stack", should we attempt to its content to StackTrace?
 			TimeEvents: jProtoLogsToOCProtoTimeEvents(jspan.Logs),
 			Links:      jProtoReferencesToOCProtoLinks(jspan.References),
 			Status:     sStatus,
 		}
 
+		if jspan.ParentSpanID() != 0 {
+			span.ParentSpanId = tracetranslator.UInt64ToByteSpanID(uint64(jspan.ParentSpanID()))
```

Comment:
Is this special-cased because we want ParentSpanId in OC format to be nil when ParentSpanID in Jaeger format is 0?

Reply:
From what I see we already have a special-case in the reverse translation [1] so this looks good to me.

[1] https://github.com/open-telemetry/opentelemetry-collector/blob/9ecbe75898683aa5316a80ae1a5cb15dfdd1651f/translator/trace/jaeger/protospan_to_jaegerproto.go#L492

Reply:
So whenever we translate from a 0 trace/span ID we have to decide if that means a 0 id or a nil id.  We've decided we feel like the default is all 0's which I like, but ParentSpanId is a special case because a nil ParentSpanId is common and correct.

There is a special case on the reverse translation back to Jaeger, but if we don't translate back to Jaeger then I believe we would want clearly flag "no parent id" with a nil instead of a 0's byte slice which might be confusing.

Another option is to go with back to my second attempt which was basically:
- Translate all 0 span/trace ids as nil
- When you translate any nil span/trace id back to Jaeger choose 0.

Reply:
I think this PR is good, no need to go back to the previous attempt.